### PR TITLE
Reorganize docs on External Login Providers

### PR DESCRIPTION
**Pull request by sofietoft**
We currently have 4 articles containing the same (or very similar!) examples of how to use external login providers for authentication users and/or members in an Umbraco CMS installation.

This PR aims to clean this up.

**To do**

* [x] Make code samples in External Login Provider article generic
* [x] Combine External Login Provider article and Auto-linking article
* [x] Test "Add Google authentication" tutorial
* [x] Test "Add Azure Active Directory authentication" tutorial

**I need help with testing the "Azure Active Directory authentication" tutorial.**

### CODE REVIEW

**Review by sofietoft — COMMENTED** [review]
Thanks a bunch for testing our the Azure AD article @nikolajlauridsen ! 🙌 

I realize now, that I forgot to mention: This article is only supposed to showcase an example for Members 😅 We already have another tutorial for Users - it's the Google one, also in this PR.

I've tried committing your suggestions. But it won't go through for some reason 🤷‍♀️ I'll keep trying!
UPDATE: Yeah. GitHub has issues 😅 I'll commit your suggestions as soon as they're up and running again!

Any chance you had a look at the other articles as well? 🤞 
Seems like the "constants" thing you've added in the Azure AD, could be used in the generic examples I've added to the External Login Provider article.

**Comment by nikolajlauridsen** [comment]
I didn't look at thee other articles no, since they had the box ticked 😅 but I can have a look at those as well 👍

File: 11/umbraco-cms/reference/security/external-login-providers.md [hunk 1]
```diff
@@ -21,45 +21,379 @@ Install an appropriate Nuget package for the provider you wish to use. Some popu
...
+using Umbraco.Cms.Web.Common.Security;
+
+namespace MyUmbracoProject.CustomAuthentication
+{
+    public class ProviderMembersExternalLoginProviderOptions : IConfigureNamedOptions<MemberExternalLoginProviderOptions>
+    {
+        public const string SchemeName = "OpenIdConnect";
+        public void Configure(string? name, MemberExternalLoginProviderOptions options)
+        {
+            if (name != "Umbraco." + SchemeName)
```
Comment (nikolajlauridsen) [inline 11/umbraco-cms/reference/security/external-login-providers.md:266]:
```suggestion
            if (name != Constants.Security.MemberExternalAuthenticationTypePrefix + SchemeName)
```

Constant here too 😄

**Review by nikolajlauridsen — COMMENTED** [review]
Just had a look at the generic one 👍, I'll check out and test the google one tomorrow 😄

File: 11/umbraco-cms/tutorials/add-google-authentication.md [hunk 2]
```diff
@@ -28,249 +28,307 @@ I'm sure a lot of content editors and implementors of your Umbraco sites would l
...
+To use an external login provider such as Google on your Umbraco CMS project, you have to implement a couple of new classes:
 
-1.  To enable a user to link their user account to an external login provider such as Google in the Umbraco Backoffice, you have to implement a custom-named configuration `BackOfficeExternalLoginProviderOptions` for users.
+* A custom-named configuration class.
+* A static extension class.
 
-    You can create a `GoogleBackOfficeExternalLoginProviderOptions.cs` file in a location of your choice.\
-    \
-    For tutorial purposes, I have created the file in `App_Code/Google_Authentication` folder.
+You can create these files in a location of your choice. In this tutorial, the files will be added to an `App_Code/Google_Authentication` folder.
```
Comment (nikolajlauridsen) [inline 11/umbraco-cms/tutorials/add-google-authentication.md:152]: I must admit that I'm really not a fan of the `App_Code/Google_Authentication` folder name, `App_Code` was very much a .Net Framework thing, and is not a thing more in modern .Net, additionally using underscores in namespaces is generally not recommended.

Maybe something along the `ExternalUserLogins/Google` or something would be better. (Do note that the namespace in the code would have to change as well)

File: 11/umbraco-cms/tutorials/add-google-authentication.md [hunk 3]
```diff
@@ -28,249 +28,307 @@ I'm sure a lot of content editors and implementors of your Umbraco sites would l
...
+{
+    public class GoogleBackOfficeExternalLoginProviderOptions : IConfigureNamedOptions<BackOfficeExternalLoginProviderOptions>
     {
-        public class GoogleBackOfficeExternalLoginProviderOptions : IConfigureNamedOptions<BackOfficeExternalLoginProviderOptions>
+        public const string SchemeName = "OpenIdConnect";
+        public void Configure(string name, BackOfficeExternalLoginProviderOptions options)
         {
-            public const string SchemeName = "OpenIdConnect";
-            public void Configure(string name, BackOfficeExternalLoginProviderOptions options)
+            if (name != "Umbraco." + SchemeName)
```
Comment (nikolajlauridsen) [inline 11/umbraco-cms/tutorials/add-google-authentication.md:170]:
```suggestion
            if (name != Constants.Security.BackOfficeExternalAuthenticationTypePrefix + SchemeName)
```

Constant here

**Review by nikolajlauridsen — COMMENTED** [review]
Only found a couple of minor things in the google ones.

I've been through all 3 articles now 👍

**Comment by nikolajlauridsen** [comment]
That looks all good to me 👍

**Comment by sofietoft** [comment]
Thanks again @nikolajlauridsen ! 💪 

I'll remove the Users part from the Azure AD article, and merge it in!

**Review by jonat123 — APPROVED** [review]

**Pull request merged**